src/ExecBackends/MemBackendUmpireImpl.hpp

```diff
@@ -0,0 +1,261 @@
+    auto& rm = umpire::ResourceManager::getInstance();
+
+    if(n>0) {
+      double* src = const_cast<double*>(p_src);
+      rm.copy(p_dest, src, n*sizeof(T));
+    }
+    return true;
+  }
+};
+
+////////////////////////////////////////////////////////////////////////////////////////
+// Transfers to/from Host C++ memory
+////////////////////////////////////////////////////////////////////////////////////////
+template<class EXECPOLDEST, class EXECPOLSRC, typename T, typename I>
+struct TransferImpl<MemBackendCpp, EXECPOLDEST, MemBackendUmpire, EXECPOLSRC, T, I>
+{
+  inline static bool do_it(T* p_dest,
+                           ExecSpace<MemBackendCpp, EXECPOLDEST>& hwb_dest,
+                           const T* p_src,
+                           const ExecSpace<MemBackendUmpire, MemBackendUmpire>& hwb_src,
+                           const I& n)
+  {
+    if(hwb_src.mem_backend().is_host()) {
+      std::memcpy(p_dest, p_src, n*sizeof(T));
+    } else {
+      assert(false && "Transfer BACKENDS(TO:Cpp-host,FROM:umpire) only supported with Umpire mem space host"); 
+      return false;
+    }
+    return true;
+  }
+};
+
+template<class EXECPOLDEST, class EXECPOLSRC, typename T, typename I>
+struct TransferImpl<MemBackendUmpire, EXECPOLDEST, MemBackendCpp, EXECPOLSRC, T, I>
+{
+  inline static bool do_it(T* p_dest,
+                           ExecSpace<MemBackendUmpire, EXECPOLDEST>& hwb_dest,
+                           const T* p_src,
+                           const ExecSpace<MemBackendCpp, EXECPOLSRC>& hwb_src,
+                           const I& n)
+  {
+    if(hwb_dest.mem_backend().is_host()) {
+      std::memcpy(p_dest, p_src, n*sizeof(T));
+    } else {
+      assert(false && "Transfer BACKENDS(TO:Umpire,FROM:Cpp-host) only supported with Umpire mem space host"); 
+      return false;
+    }
+    return true;
+  }
+};
+
+////////////////////////////////////////////////////////////////////////////////////////
+// Transfers to/from CUDA memory
+////////////////////////////////////////////////////////////////////////////////////////
+#ifdef HIOP_USE_CUDA
+template<class EXECPOLDEST, class EXECPOLSRC, typename T, typename I>
+struct TransferImpl<MemBackendCuda, EXECPOLDEST, MemBackendUmpire, EXECPOLSRC, T, I>
+{
+  inline static bool do_it(T* p_dest,
+                           ExecSpace<MemBackendCuda, EXECPOLDEST>& hwb_dest,
+                           const T* p_src,
+                           const ExecSpace<MemBackendUmpire, EXECPOLSRC>& hwb_src,
+                           const I& n)
+  {
+    if(hwb_src.mem_backend().is_device()) {
+      return cudaSuccess == cudaMemcpy(p_dest, p_src, n*sizeof(T), cudaMemcpyDeviceToDevice);
+    } else {
+      if(hwb_src.mem_backend().is_host()) {
+        return cudaSuccess == cudaMemcpy(p_dest, p_src, n*sizeof(T), cudaMemcpyHostToDevice);
+      } else {
+        assert(false && "Transfer BACKENDS(TO:Cuda,FROM:umpire) not supported with Umpire mem space 'um'");
+        return false;
```

Comment:
Might be missing something here... I would expect that this can be supported, but on the other hand, `um` memory is automatically transferred between device/host as needed, so users would be doing something quite weird to invoke this part of HiOp :)